biz.aQute.bndlib/src/aQute/bnd/osgi/FileResource.java

```diff
@@ -51,7 +52,7 @@ public String toString() {
 	}
 
 	public void write(OutputStream out) throws Exception {
-		copy(this, out);
+		Files.copy(file.toPath(), out);
```

Comment:
Is there a problem with the current implementation? What does this change fix?

So why leave the old method behind then? It is not API.

It is used by ZipResource.write. But since is not API, that method can be moved to ZipResource.